### PR TITLE
jupyter/toc: fix #4166

### LINKED ISSUE #4166: jupyter/TOC: Unable to get property 'split' of undefined or null reference
this must be in `smc-webapp/jupyter/contents.ts` → `parse_cell_heading`

---

chrome: `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.140 Safari/537.36 Edge/18.17763`

smc_git_rev: b612379cd2218f31d34a2eb1eab1255c436ffbb2

Date: 2019-10-15 08:14:46.780537

`Unable to get property 'split' of undefined or null reference`

```
TypeError: Unable to get property 'split' of undefined or null reference
   at Anonymous function (https://cocalc.com/static/smc-75b1a24eaac3878d8530.cacheme.js?75b1a24eaac3878d8530:5:2758226)
   at Anonymous function (https://cocalc.com/static/smc-75b1a24eaac3878d8530.cacheme.js?75b1a24eaac3878d8530:5:2758192)
   at dt.prototype.__iterate (https://cocalc.com/static/0-75b1a24eaac3878d8530.cacheme.js?75b1a24eaac3878d8530:6:367178)
   at forEach (https://cocalc.com/static/0-75b1a24eaac3878d8530.cacheme.js?75b1a24eaac3878d8530:6:391136)
   at t.parse_headings (https://cocalc.com/static/smc-75b1a24eaac3878d8530.cacheme.js?75b1a24eaac3878d8530:5:2757803)
   at t.prototype.update_contents (https://cocalc.com/static/smc-75b1a24eaac3878d8530.cacheme.js?75b1a24eaac3878d8530:5:2756729)
   at m (https://cocalc.com/static/0-75b1a24eaac3878d8530.cacheme.js?75b1a24eaac3878d8530:6:662061)
   at b (https://cocalc.com/static/0-75b1a24eaac3878d8530.cacheme.js?75b1a24eaac3878d8530:6:662273)
   at g (https://cocalc.com/static/0-75b1a24eaac3878d8530.cacheme.js?75b1a24eaac3878d8530:6:662184)
   at e._wrapper (https://cocalc.com/static/smc-75b1a24eaac3878d8530.cacheme.js?75b1a24eaac3878d8530:5:698226)
```

### PR DESCRIPTION
**Pull request by haraldschilly**
# Description

well, basically `cell.get("input")` could return null or undefined ...

# Testing Steps
1. toc shows up

# Relevant Issues

### [Checklist](https://github.com/sagemathinc/cocalc/wiki/PR-Checklist):
- [ ] No debugging console.log messages.
- [ ] All new code is actually used.
- [ ] Non-obvious code has some sort of comments.

Front end:
- [ ] Restart at least one project and check its `~/.smc/local_hub/local_hub.log`
- [ ] Completely restart Webpack with `./w` in `/src`
- [ ] Completely restart the hub by killing and restarting `./start_hub.py` in `/src/dev/project`
- [ ] Screenshots if relevant.

### CODE REVIEW

**Comment by williamstein** [comment]
I think it's shorter, better and more robust code to just not call `parse_cell_heading` if input isn't defined.  That way you don't need comments like `// returning undefined means there is no heading in markdown`.   You're rewritten this code with exactly the sort of old defensive style we had to do before typescript came along.  The whole advantage of typescript is that you don't have to write code like this. 

In short, just replace this whole patch by:
```
    const input = cell.get("input");
    if (input == null) return;
    const { level, value } = parse_cell_heading(input);
```
